cmd/cluster/core/destroy.go

```diff
@@ -129,17 +131,41 @@ func DestroyCluster(ctx context.Context, hostedCluster *hyperv1.HostedCluster, o
 				o.Log.Error(err, "Failed to get hosted cluster", "namespace", o.Namespace, "name", o.Name)
 				return false, nil
 			}
-			done := len(hostedCluster.Finalizers) == 1 && hostedCluster.Finalizers[0] == destroyFinalizer
+			done := len(hostedCluster.Finalizers) == 1 && hostedCluster.Finalizers[0] == DestroyClusterFinalizer
 			return done, nil
 		}, clusterDeleteCtx.Done())
 		if err != nil {
 			return fmt.Errorf("hostedcluster wasn't finalized, aborting delete: %w", err)
 		}
 	}
 
-	// Destroy additional resources which are specific to the current platform
-	if err := destroyPlatformSpecifics(ctx, o); err != nil {
-		return err
+	// Destroy additional resources which are specific to the current platform (Skip and just remove finalizer with HCI)
+	hci := hyperv1.HostedClusterInfrastructure{
+		ObjectMeta: corev1.ObjectMeta{
+			Name:      o.InfraID,
+			Namespace: o.Namespace,
+		},
+	}
+	hciFound := false
+	if err := c.Get(ctx, client.ObjectKeyFromObject(&hci), &hci); err == nil {
```

Comment:
can we please invert this check as do err != convention?

Reply:
done